parachain/primitives/core/src/rewards.rs

```diff
@@ -0,0 +1,31 @@
+use frame_support::traits::{Currency, ExistenceRequirement::KeepAlive};
+use frame_system::Config;
+use sp_runtime::traits::Zero;
+use sp_std::marker::PhantomData;
+
+
+pub trait RewardRelayer<AccountId, Balance> {
+	fn pay_relayer(source: &AccountId, relayer: &AccountId, reward: Balance);
+}
+
+pub struct InstantRewards<T, C>(PhantomData<(T, C)>);
+
+impl<T, C> RewardRelayer<T::AccountId, C::Balance> for InstantRewards<T, C>
+where
+	T: Config,
+	C: Currency<T::AccountId>
+{
+	fn pay_relayer(source: &T::AccountId, relayer: &T::AccountId, reward: C::Balance) {
+		if reward.is_zero() {
+			return;
+		}
+
+		let _ = C::transfer(
```

Comment:
Note that transfer will be logged in Polkadot explorers, so it will allow users to identify the relayer easily, and also allow relayers to see where the rewards came from. Fees in substrate are normally not transferred, but rather withdrawn and then deposited, which means that fees do not show up as transfers in explorers.

I have a slight opinion that rewards/fees should not show up as transfers, because such a transfer was not the user intention.

Reply:
ah good catch! Will add a fix for it.

Reply:
done.